### DISCUSSION TITLE
com.google.android.gms.ads.AdActivity error in android studio cordova project

### DISCUSSION THREAD

**Discussion by superdev40** [admob-plus/admob-plus#477] [Q&A]
Hello, when I install admobplus in my cordova project under android studio with: 
`cordova plugin add admob-plus-cordova --save --variable APP_ID_ANDROID = ca-app-pub-xxx ~ xxx --variable APP_ID_IOS = ca-app-pub -xxx ~ xxx.`
 I end up with the `com.google.android.gms.ads.AdActivity` line in red and the error tells me :

`Class referenced in the manifest, com.google.android.gms.ads.AdActivity, was not found in the project or the libraries`

Do you understand why there is this error? Thanks for your help

**Comment by superdev40** [comment]
Anglais

I finally managed to correct the problem by including` classpath "com.google.android.gms: play-services-ads: 20.5.0"` in the dependencies of the gradle

**Reply by keev-studio** [reply]
can you please elaborate on this, I am facing the same problem

**Reply by superdev40** [reply]
Hello, what are you trying to do?

**Reply by keev-studio** [reply]
I am getting this error for some reason:
`Uncaught (in promise) {
  "Code": 3,
  "Message": "No ad config.",
  "Domain": "com.google.android.gms.ads",
  "Cause": "null",
  "Response Info": {
    "Response ID": "null",
    "Mediation Adapter Class Name": "",
    "Adapter Responses": []
  }
}`

**Reply by superdev40** [reply]
"No ad config"

Have you put you ID admod in the file androidManifest.xml ? 

```
 <meta-data
            android:name="com.google.android.gms.ads.APPLICATION_ID"
            android:value="ca-app-pub-ADMODID" /> 
```

**Reply by keev-studio** [reply]
Hi, thanks for your reply. I have the above mentioned line in my androidManifest.xlm but still getting the same error. In Android Studio logcat its producing this error

2022-02-19 16:21:45.799 15767-16060/es.keev.audioknihy.lite E/FA: Missing google_app_id. Firebase Analytics disabled. See https://goo.gl/NAOOOI
(but I dont use Firebase in my app at all)

P.S. in Chrome inspect the error is defined as:
Uncaught (in promise) {
  "Code": 3,
  "Message": "No ad config.",
  "Domain": "com.google.android.gms.ads",
  "Cause": "null",
  "Response Info": {
    "Response ID": "null",
    "Mediation Adapter Class Name": "",
    "Adapter Responses": []
  }
}

I dont know what to do ...

**Reply by superdev40** [reply]
Did you add 

`apply plugin: 'com.google.gms.google-services'`

to the bottom of your modules build.gradle?
